src/rpc/claimtrie.cpp

```diff
@@ -183,11 +183,14 @@ UniValue getvalueforname(const UniValue& params, bool fHelp)
     std::string sValue;
     if (!getValueForClaim(claim.outPoint, sValue))
         return ret;
+
     ret.push_back(Pair("value", sValue));
     ret.push_back(Pair("txid", claim.outPoint.hash.GetHex()));
     ret.push_back(Pair("n", (int)claim.outPoint.n));
-    ret.push_back(Pair("amount", claim.nAmount));
-    ret.push_back(Pair("effective amount", claim.nEffectiveAmount)); 
+    ret.push_back(Pair("amount", ValueFromAmount(claim.nAmount)));
+    claimsForNameType claimsForName = pclaimTrie->getClaimsForName(name);
```

Comment:
You could, it would functionally be calling getClaimsForName and then performing the calculation. So was thinking its not very efficient when calculating effective amounts for multiple claims as we are doing here. 

Unless there's some better way to obtain the effective amount ?